src/coordinator.js

```diff
@@ -5,7 +5,19 @@ import './environment';
 import logger from './utils/logger';
 import { raceTo } from './utils/lifecycle';
 
-const WORKER_COUNT = os.cpus().length - 1 || 1;
+export function getWorkerCount(cpus) {
+  const realCount = os.cpus().length;
+
+  if (cpus === 0) {
+    return realCount - 1 || 1;
+  }
+
+  if (cpus > realCount) {
+    logger.info(`Specified CPU count of ${cpus} is bigger than OS reported count of ${realCount}`);
+  }
+
+  return cpus - 1;
```

Comment:
I think making the "-1" behavior implicit combined with s configurable CPU count is problematic.

What if "cpus" was a function that took `realCount` and returned a number? The default would do the -1 logic; hypernova would throw if it returned a non-number; and it'd be easy to do `() => 3` or `X => X - 1`